plugins/sqlfluff-templater-dbt/sqlfluff_templater_dbt/templater.py

```diff
@@ -596,36 +289,11 @@ def make_template(in_str):
             [],
         )
 
-    @contextmanager
-    def connection(self):
-        """Context manager that manages a dbt connection, if needed."""
-        # We have to register the connection in dbt >= 1.0.0 ourselves
-        # In previous versions, we relied on the functionality removed in
-        # https://github.com/dbt-labs/dbt-core/pull/4062.
-        if DBT_VERSION_TUPLE >= (1, 0):
-            if not self.connection_acquired:
-                adapter = get_adapter(self.dbt_config)
-                adapter.acquire_connection("master")
-                adapter.set_relations_cache(self.dbt_manifest)
-                self.connection_acquired = True
-            yield
-            # :TRICKY: Once connected, we never disconnect. Making multiple
-            # connections during linting has proven to cause major performance
-            # issues.
-        else:
-            yield
-
 
 class SnapshotExtension(StandaloneTag):
     """Dummy "snapshot" tags so raw dbt templates will parse.
 
-    Context: dbt snapshots
-    (https://docs.getdbt.com/docs/building-a-dbt-project/snapshots/#example)
-    use custom Jinja "snapshot" and "endsnapshot" tags. However, dbt does not
-    actually register those tags with Jinja. Instead, it finds and removes these
-    tags during a preprocessing step. However, DbtTemplater needs those tags to
-    actually parse, because JinjaTracer creates and uses Jinja to process
-    another template similar to the original one.
+    For more context, see sqlfluff-templater-dbt.
```

Comment:
Note the new templater is less than half the number of lines as the old one. A lot of cruft has been removed, and the osmosis dbt wrappers simplify things as well.